src/test/unit_tests/test_bucketspace.js

```diff
@@ -41,6 +41,169 @@ const DEFAULT_FS_CONFIG = {
+                const obj_nsr = { resource: dummy_nsr, path: bucket };
+                const create_bucket_options = {
+                    name: bucket,
+                    namespace: {
+                        read_resources: [obj_nsr],
+                        write_resource: obj_nsr
+                    }
+                };
+                if (process.env.NC_CORETEST) create_bucket_options.owner = account_name;
+                await rpc_client.bucket.create_bucket(create_bucket_options);
+            } else {
+                const res = await s3_client.createBucket({ Bucket: bucket });
+                console.log('created bucket via s3', account_name, bucket, res);
+            }
+        }
+    });
+
+    mocha.after(async function() {
+        this.timeout(60000); // eslint-disable-line no-invalid-this
+        for (const account_name of Object.keys(accounts)) {
+            const { bucket } = accounts[account_name];
+            await rpc_client.bucket.delete_bucket({ name: bucket });
+            await rpc_client.account.delete_account({ email: `${account_name}@noobaa.com` });
+        }
+        await fs_utils.folder_delete(tmp_fs_root);
+    });
+
+    mocha.it('list buckets - each account can list only its own bucket - no bucket policies applied ', async function() {
+        for (const account of Object.values(accounts)) {
+            const { s3_client } = account;
+            const res = await s3_client.listBuckets({});
+            const buckets = res.Buckets.filter(bucket => bucket.Name !== 'first.bucket');
+            assert.equal(buckets.length, 1);
+            assert.equal(buckets[0].Name, account.bucket);
+        }
+    });
+    mocha.it('account1 - set allow all policy, all account can list bucket1', async function() {
+        // Give account access to the required buckets
+        const public_bucket = accounts.account1.bucket;
+        await Promise.all(
+            [public_bucket]
+            .map(bucket => test_utils.generate_s3_policy('*', bucket, ['s3:*']))
+            .map(generated =>
+                rpc_client.bucket.put_bucket_policy({
+                    name: generated.params.bucket,
+                    policy: generated.policy,
+                })
+            )
+        );
+        for (const account of Object.values(accounts).slice(1)) {
+            const { s3_client } = account;
+            const res = await s3_client.listBuckets({});
+            const buckets = res.Buckets.filter(bucket => bucket.Name !== 'first.bucket' && bucket.Name !== account.bucket);
+            assert.equal(buckets.length, 1);
+            assert.equal(buckets[0].Name, public_bucket);
+        }
+
+        // delete bucket policy
+        await rpc_client.bucket.put_bucket_policy({
+            name: public_bucket,
+            policy: '',
+        });
+    });
+
+    mocha.it('account2 - set allow all only account1, account1 should list bucket2 but account3 shouldnt', async function() {
```

Comment:
This test seems complicated to understand, so try to simplify it (remember that it might fail to someone who touches a small part of the code and needs to understand how to debug it...).

Reply:
I added comments